### DISCUSSION TITLE
Configure multiple workers when using Pytest?

### DISCUSSION THREAD

**Discussion by pg** [schemathesis/schemathesis#2509] [Q&A] [answered]
All of our pytest tests (~170) complete under ~5s, but now with Schemathesis tests added, our test suite takes ~4-5m. I know it's generating a whole lot more tests for us, but I was hoping for some ways to minimize the hit to our overall suite timing. 

Is there a way to configure the `workers` setting somewhere (via the schemathesis object or in pyproject) when we run our entire suite with pytest? I'm hoping we can find a way to speed up the suite.

We're testing a FastAPI app, so maybe there are other test optimizations we should consider. Is it unnecessarily setting up and tearing down the app too frequently?

**Comment by Stranger6667** [comment, marked as answer]
Hi!

> Is there a way to configure the workers setting somewhere (via the schemathesis object or in pyproject) when we run our entire suite with pytest?

`pytest-xdist` should work - it is used in Schematheis' own test suite and yielded great results at the time it was introduced.

> We're testing a FastAPI app, so maybe there are other test optimizations we should consider. Is it unnecessarily setting up and tearing down the app too frequently?

What I often observe is that frequent cleanups don't bring much in terms of finding novel bugs (and don't give much trouble with reproducing bugs either), but have a high cost. Therefore I'd suggest reducing setup / teardown by putting the app into a separate fixture with the "session" scope. 

To avoid the network stack you can use `schemathesis.from_asgi` loader, so it will communicate to the ASGI app directly, which is faster. See this [doc section](https://schemathesis.readthedocs.io/en/stable/python.html#asgi-wsgi-support) for more info. 

There is also a [lazy way of schema loading](https://schemathesis.readthedocs.io/en/stable/python.html#lazy-loading), which may be helpful too

Please, let me know if you'd like me to elaborate
